### PR TITLE
Fix RPC nonfatal errors

### PR DESCRIPTION
**Pull request by jzila**
Method and protocol missing errors are no longer fatal to the transport.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
lgtm

although a follow-up commit with tests exercising this would be good.

File: protocol_test.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func TestBrokenCall(t *testing.T) {
 	defer endTest(t, conn, listener)
 
 	err := cli.Broken()
-	require.Error(t, err, "Called nonexistent method, expected error")
+	require.EqualError(t, err, "method 'broken' not found in protocol 'test.1.testp'")
```
Comment (maxtaco) [inline protocol_test.go]: john can you add an RPC that works after this call, to make sure there wasn't an EOF?

Comment (jzila) [inline protocol_test.go]: Just did that :)

**Comment by maxtaco** [comment]
:+1:  thanks for the speedy fix!

**Pull request merged**